nautobot/ipam/forms.py

```diff
@@ -734,12 +744,14 @@ def save(self, *args, **kwargs):
             else:
                 interface.parent.primary_ip6 = ipaddress
             interface.parent.save()
-        elif interface and ipaddress.address.version == 4 and interface.parent.primary_ip4 == ipaddress:
-            interface.parent.primary_ip4 = None
-            interface.parent.save()
-        elif interface and ipaddress.address.version == 6 and interface.parent.primary_ip6 == ipaddress:
-            interface.parent.primary_ip6 = None
-            interface.parent.save()
+
+        # Save the `original_assigned_object.parent` if `_primary_ip_unset_by_form` was set in `clean()`
+        primary_ip_unset_by_form = getattr(self.instance, "_primary_ip_unset_by_form", False)
+        if primary_ip_unset_by_form:
+            ip_version = self.instance.address.version
```

Comment:
Any reasoning for using `self.instance` instead of the `ipaddress` object like the other code here?

Reply:
I need an adult.